python/lsst/pipe/base/connectionTypes.py

```diff
@@ -57,6 +57,14 @@ class BaseConnection:
     doc: str = ""
     multiple: bool = False
 
+    deprecated: str | None = dataclasses.field(default=None, kw_only=True)
+    """A description of why this connectoin is deprecated, including the
```

Comment:
It seems we are not even documenting `doc` above, but it seems weird to have this comment here rather than in the class level documentation like the others. Can you either move the doc to the docstring, or move the other docs to the objects? Which way does sphinx prefer? I do know that python does nothing with docstrings under the object, so interactively they just disappear.